src/paymentservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/paymentservice/BamboraClientImpl.java

```diff
@@ -0,0 +1,184 @@
+package ca.bc.gov.open.pssg.rsbc.dps.paymentservice;
+
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.UUID;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import ca.bc.gov.open.pssg.rsbc.dps.paymentservice.PaymentServiceConstants.BamboraTransType;
+import ca.bc.gov.open.pssg.rsbc.dps.paymentservice.exception.PaymentServiceException;
+
+/**
+ * Bambora Client class Implementation
+ * 
+ * May be expanded in the future to include other types of Bambora transactions. 
+ * 
+ * @author smillar
+ *
+ */
+public class BamboraClientImpl implements PaymentClient {
+
+	private URL hostedPaymentURL = null;
+	private String merchantId = null;
+	private String hashKey = null;
+	private int minutesToExpire;
+	private String correlationId = null;
+	
+	public URL getHostedPaymentServiceURL() {
+		return hostedPaymentURL;
+	}
+
+	public String getMerchantId() {
+		return merchantId;
+	}
+
+	public URL getHostedPaymentURL() {
+		return hostedPaymentURL;
+	}
+
+	public int getMinutesToExpire() {
+		return minutesToExpire;
+	}
+	
+	public String getCorrelationId() {
+		return correlationId;
+	}
+
+	// constructor
+	public BamboraClientImpl(URL hostedPaymentURL, String merchantId, String hashKey, int minutesToExpire) throws PaymentServiceException {
```

Comment:
should we validate the input params?

Reply:
I like to keep my validation in the controller which is already set up to provide validation. 
If you look at ca.bc.gov.open.pssg.rsbc.dps.paymentservice.CalculateSinglePayment, Line: 81 you will see the validation for this constructors parameters.